### ISSUE TITLE
[@typescript-eslint/no-explicit-any] Error when specifying `ignoreRestArgs` option

### ISSUE THREAD

**Issue by doniyor2109** [typescript-eslint/typescript-eslint#1795]
**Repro**
Here is repo:
https://github.com/doniyor2109/demo-error-for-typescript-eslint-no-exlicit-any

```JSON
{
  "rules": {
    "@typescript-eslint/no-explicit-any": [
      "error",
      {
        "ignoreRestArgs": true
      }
    ]
  }
}
```

```TS
	type Request = any;
```

**Expected Result**
Should report about usage of `any`.
**Actual Result**
Throws error: `TypeError: Cannot read property 'type' of null`

**Additional Info**
Stack trace
```
node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-explicit-any.js:71:26)
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.25.0`|
| `@typescript-eslint/parser`        | `2.25.0`|
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.2.0`|
| `npm`                              | `6.9.0` |
